tests/std/tests/P0408R7_efficient_access_to_stringbuf_buffer/test.cpp

```diff
@@ -0,0 +1,177 @@
+        stream.str(init_value);
+        assert(stream.view() == init_value);
+        assert(stream.str(init_value.get_allocator()) == init_value);
+        assert(stream.str() == string_view{init_value});
+    }
+};
+
+template <typename Test>
+void run_test_util() {
+    Test test{};
+    test(empty_string);
+    test(small_string);
+    test(large_string);
+}
+
+template <template <typename> typename Test>
+void run_test() {
+    run_test_util<Test<stringstream>>();
+    run_test_util<Test<istringstream>>();
+    run_test_util<Test<ostringstream>>();
+}
+
+template <typename T>
+struct counting_allocator {
+    using value_type = T;
+
+    std::allocator<T> allocator{};
+    std::size_t count{0};
+
+    T* allocate(std::size_t n) {
+        count++;
+        return allocator.allocate(n);
+    }
+
+    void deallocate(T* p, std::size_t n) {
+        allocator.deallocate(p, n);
+    }
+
+    counting_allocator() {}
+
+    template <typename U>
+    explicit counting_allocator(const counting_allocator<U>& c) : count(c.count) {}
+};
+
+template <typename T, typename U>
+bool operator==(const counting_allocator<T>& c1, const counting_allocator<U>& c2) {
+    return c1.allocator == c2.allocator;
+}
+
+using counting_string        = std::basic_string<char, std::char_traits<char>, counting_allocator<char>>;
+using counting_stringstream  = std::basic_stringstream<char, std::char_traits<char>, counting_allocator<char>>;
+using counting_istringstream = std::basic_istringstream<char, std::char_traits<char>, counting_allocator<char>>;
+using counting_ostringstream = std::basic_ostringstream<char, std::char_traits<char>, counting_allocator<char>>;
+
+template <typename Stream>
+struct test_counting_allocator {
+    void operator()(counting_string s) {
+        Stream stream{move(s)};
+        s = move(stream).str();
+        stream.str(move(s));
+        if constexpr (!std::is_same_v<Stream, counting_istringstream>) {
+            assert(stream.rdbuf()->get_allocator().count == 1);
+        }
+    }
+};
+
+template <template <typename> typename Test>
+void run_counting_test() {
+    run_test_util<Test<counting_stringstream>>();
+    run_test_util<Test<counting_istringstream>>();
+    run_test_util<Test<counting_ostringstream>>();
+}
+
+template <const char* str>
+void test_iterator_dereference_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    (void) *iterator; // cannot dereference invalid iterator
+}
+
+template <const char* str>
+void test_iterator_operator_arrow_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    (void) iterator.operator->(); // cannot dereference invalid iterator
+}
+
+template <const char* str>
+void test_iterator_increment_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    ++iterator; // cannot increase invalid iterator
+}
+
+template <const char* str>

```

Comment:
I think this would be better as an ordinary runtime parameter (you don't need distinct function pointers here since this one isn't address-taken)

Reply:
Well, yes. `death_test_executive::add_death_tests` accepts an array of function pointers.